lib/Basics/ResultT.h

```diff
@@ -282,6 +293,49 @@ class ResultT {
       : _result(std::move(result)), _val(std::move(val_)) {}
 };
 
+template<typename T>
+struct ResultTSerializer
+    : std::variant<std::reference_wrapper<const arangodb::Result>,
+                   std::reference_wrapper<T>> {};
+template<typename Inspector, typename T>
+auto inspect(Inspector& f, ResultTSerializer<T>& x) {
+  return f.variant(x).unqualified().alternatives(
+      arangodb::inspection::type<std::reference_wrapper<T>>("value"),
+      arangodb::inspection::type<
+          std::reference_wrapper<const arangodb::Result>>("error"));
+}
+template<typename T>
+struct ResultTDeserializer : std::variant<arangodb::Result, T> {};
+template<typename Inspector, typename T>
+auto inspect(Inspector& f, ResultTDeserializer<T>& x) {
+  return f.variant(x).unqualified().alternatives(
+      arangodb::inspection::type<T>("value"),
+      arangodb::inspection::type<arangodb::Result>("error"));
+}
```

Comment:
These are only helpers for the ResultT inspect function. Can we move them into a separate namespace (e.g., `detail`)?